src/libDirectoryService/FinalBlockPostProcessing.cpp

```diff
@@ -170,6 +170,10 @@ void DirectoryService::ProcessFinalBlockConsensusWhenDone() {
         << m_mediator.m_txBlockChain.GetLastBlock().GetHeader().GetBlockNum() +
                1
         << "] FINISH WRITE STATE TO DISK");
+    if (m_mediator.m_dsBlockChain.GetLastBlock().GetHeader().GetBlockNum() <
```

Comment:
Should we add a check for `ENABLE_ACCOUNTS_POPULATING` to make this part more efficient?